lib/internal/url.js

```diff
@@ -133,6 +133,12 @@ function onParseProtocolComplete(flags, protocol, username, password,
   if ((s && !newIsSpecial) || (!s && newIsSpecial)) {
     return;
   }
+  if (protocol === 'file:' && (username || password || port > -1)) {
```

Comment:
`port` can only be `>= 0` or `undefined`, since the C++ layer sets `port` to undefined if it is `-1`, and you should probably check `ctx.username` and `ctx.password` instead of `username` and `password` since IIRC the C++ layer doesn't pass those into the callback if it's unneeded.